### PR TITLE
added "black" to requirements.txt 

### LINKED ISSUE #215: Add `black` to `requirements.txt` for developers
**Is your feature or enhancement request related to a problem or limitation? Please describe**

_(original issue: #111 submitted by @t1m0thyj)_

**Describe your enhancement idea**

We recommend that `black` is used for linting, but we do not have it as part of the `requirements.txt` file.

Request: Add `black` to `requirements.txt`.

### PR DESCRIPTION
**Pull request by Myrausman**
**What It Does**
added "black" to requirements.txt
Resolved #215 

**How to Test**

**Review Checklist**
I certify that I have:
- [X] tested my changes
- [X] added/updated automated tests
- [ ] updated the changelog
- [X] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Comment by Myrausman** [comment]
@t1m0thyj , I added "black"  to requirements.txt, but I'm not sure about the build failure. Can you please tell why ? Thank you!

File: requirements.txt [hunk 1]
```diff
@@ -27,9 +27,11 @@ Unidecode==1.2.0
...
+black
 -e ./src/core
 -e ./src/zos_console
 -e ./src/zos_files
 -e ./src/zos_jobs
 -e ./src/zos_tso
--e ./src/zosmf
\ No newline at end of file
+-e ./src/zosmf
+black
```
Comment (t1m0thyj) [inline requirements.txt:37]: Please remove this 2nd instance of "black", we should only need it once in requirements.txt

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
@Myrausman Thanks for your contribution to the Zowe Python SDK 🙂 

In requirements.txt, could you please update the `typing-extensions` version to `4.0.1`? This should hopefully fix the version incompatibility that's causing the builds to fail.

**Review by traeok — APPROVED** [review]
LGTM, thanks @Myrausman!

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**